replica/database.cc

```diff
@@ -531,6 +531,20 @@ void backlog_controller::update_controller(float shares) {
     _scheduling_group.set_shares(shares);
 }
 
+void compaction_controller::update_controller(float shares) {
+    _computed_shares = shares;
+    if (_max_shares > 0.0f && shares > _max_shares) {
+        shares = _max_shares;
+    }
```

Comment:
this is not really the maximum shares, as the maximum as defined in the control points is 1000. Maybe we should update control point and restore the default value when max shares is disabled.

Reply:
@raphaelsc Done. 

One thing to note is that, with the new version, when max_shares is set, it will reduce the shares output for all existing values and not just for values in  [max_shares, 1000].

For example, with {{0.0, 50}, {1.5, 100} , {30, 1000}}, if backlog = 20, output is approx. ≈ 684
If we set max_shares = 800, the output for backlog of 20 is ≈ 554. This is okay right?